### PR TITLE
Adds stealth modifier as mutation property

### PR DESCRIPTION
**Pull request by Siyalatas**
#### Summary
```SUMMARY: Features "Adds stealth modifier as JSON-ized mutation property"```

#### Purpose of change
1 - Fix bug with Crafty mutation, which effectively does nothing for player's visibility, since `player:visibility()`, where it is used, is never even called outside of lua bindings.
2 - JSON-ize the stealth contribution of a mutation to allow for more varied stealth-enhancing mutations

#### Describe the solution
Made the range at which monsters see the player affected by a mutation property `stealth_modifier`. The value of this mod is subtracted, as a percentage, from the player's visibility rating, and will currently not allow for values under 60% visibility range (which is 40 `stealth_modifier`). `stealth_modifier` of 40 is added to the Crafty mutation, as it was originally intended to function with this value.

#### Describe alternatives you've considered
This is my first 'real' contribution to the game, and I'm not very familiar with C++. I used a bit more conversions between ints and floats than I'd like (mostly to avoid having to write a host of new functions to handle getting int values from mutations), and I would certainly like some feedback regarding best practices. 

The changes have been tested and work properly.

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -11321,12 +11321,10 @@ int player::visibility( bool, int ) const
...
         return 0;
     }
-    if( has_trait( trait_CRAFTY ) ) {
-        return 60;
-    }
     // @todo:
     // if ( dark_clothing() && light check ...
-    return 100;
+    int stealth_modifier = std::floor( mutation_value( "stealth_modifier" ) );
+    return clamp( 100 - stealth_modifier, 60, 160 );
```
Comment (AMurkin) [inline src/player.cpp]: `clamp( x, 60, 160 )` is 110 +/- 50. Is this as intended?

Comment (Siyalatas) [reply src/player.cpp]: It wasn't, I made a blunder; fixed now

Thanks

**Comment by Mecares** [comment]
So a mutant with pointy ears in camo clothes gets easier spotted than a normal human in a clown suit?

**Comment by Siyalatas** [comment]
> So a mutant with pointy ears in camo clothes gets easier spotted than a normal human in a clown suit?

No no no, you got it all wrong old chap. A mutant with **round** ears in a clown suit is *harder* to spot than a normal human in camo clothes.

Eventually we'll probably have clothes modify visibility, but all in due time, and not in this PR. For now I just wanted to fix a bug, and try to make it possible for a guy on the forums to implement stealthy mutations that aren't "Crafty".

File: src/creature.cpp [hunk 2]
```diff
@@ -188,7 +188,10 @@ bool Creature::sees( const tripoint &t, bool is_player ) const
         }
         if( is_player ) {
             // Special case monster -> player visibility, forcing it to be symmetric with player vision.
-            return range >= wanted_range &&
+            const float player_visibility_factor = ( g->u.visibility() ) / 100.0f;
+            int adj_range = std::floor( (float) range * player_visibility_factor );
```
Comment (ZhilkinSerg) [inline src/creature.cpp]: Please replace C-style `(T)` cast with `static_cast<T>`.

File: src/creature.cpp [hunk 3]
```diff
@@ -188,7 +188,10 @@ bool Creature::sees( const tripoint &t, bool is_player ) const
         }
         if( is_player ) {
             // Special case monster -> player visibility, forcing it to be symmetric with player vision.
-            return range >= wanted_range &&
+            const float player_visibility_factor = ( g->u.visibility() ) / 100.0f;
```
Comment (BevapDin) [inline src/creature.cpp]: The outher pair of `()` is not needed.

File: src/creature.cpp [hunk 4]
```diff
@@ -188,7 +188,10 @@ bool Creature::sees( const tripoint &t, bool is_player ) const
...
+            int adj_range = std::floor( static_cast<float>( range ) * player_visibility_factor );
```
Comment (BevapDin) [inline src/creature.cpp]: The explicit cast is not needed. The second operand is a `float`, so the first will implicitly casted to one as well.

**Comment by ZhilkinSerg** [comment]
Jenkins rebuild

**Comment by ZhilkinSerg** [comment]
Jenkins rebuild.

**Comment by ZhilkinSerg** [comment]
> Soo I'm gonna say you need to astyle creature.cpp and mutation.h?

There were stray whitespaces - I've removed them.

**Pull request merged**